### PR TITLE
Show that module is loading in UI

### LINKED ISSUE #365: Show user that configuration is still loading when it takes time
**Is your feature request related to a problem? Please describe.**
Right now you can specify a configuration and you will receive that is accepted. After reloading though you will see the old configuration and this is due to the modules being stuck on building the prefix tree which might take even minutes!

**Describe the solution you'd like**
There needs to be a progress bar that actually let's user know that we are currently loading the new configuration and that it is not stuck on an unknown behavior or that it didn't pick up the new configuration at all.

### PR DESCRIPTION
**Pull request by vkotronis**
### Description of PR

What component(s) does this PR affect?

- [X] Back-End (Database, Detection/Configuration/etc. Microservices)
- [X] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Detection/Configuration/etc. Microservices)
- [ ] Front-End (Flask, API, UI, etc)
- [ ] Monitor (RIPE RIS, BGPStream RV/RIS/CAIDA, etc.)
- [ ] Docs (incl. wiki)
- [ ] Build System

### Related Issue

Resolves #365 

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: backend/core/database.py [hunk 1]
```diff
@@ -509,6 +535,29 @@ def config_request_rpc(self):
                 while self.rules is None:
                     self.connection.drain_events()
 
+        def signal_loading(self, status=False):
```
Comment (slowr) [inline backend/core/database.py:538]: database can directly call the query, no need to send a message to itself right?

Comment (vkotronis) [reply backend/core/database.py:520]: fixed

**Comment by vkotronis** [comment]
> LGTM, maybe we will need to disable some buttons when loading? or making a configuration change while the previous is loading (or throw a prompt to warn the user)

Yes I will work on this to make sure that the conf is not touched during loading.

**Comment by vkotronis** [comment]
> Edit:
> Copy the login from `overview` to `system` page as well
@slowr what do you mean here?

**Comment by vkotronis** [comment]
**BEFORE CONF CHANGE**
Screenshot_2020-05-06 ARTEMIS - Overview
Screenshot_2020-05-06 ARTEMIS - System

**IMPLEMENTING CONF CHANGE**
Screenshot_2020-05-06 ARTEMIS - System(2)
Screenshot_2020-05-06 ARTEMIS - Overview(1)

*AFTER CONF CHANGE ROLLOUT**
Screenshot_2020-05-06 ARTEMIS - Overview(2)
Screenshot_2020-05-06 ARTEMIS - System(3)

File: backend/core/database.py [hunk 2]
```diff
@@ -509,6 +518,34 @@ def config_request_rpc(self):
...
```
Comment (slowr) [inline backend/core/database.py:521]: Do you think it would make sense to move the function to utility?

**Comment by slowr** [comment]
> > Edit:
> > Copy the login from `overview` to `system` page as well
> 
> @slowr what do you mean here?

it was on the draft version that didn't have the loading on the system. all good now.

File: backend/core/database.py (hunk 2 above)
Comment (vkotronis) [reply backend/core/database.py:521]: Good idea, I will move this.

Comment (vkotronis) [reply backend/core/database.py]: Done, resolving.

**Comment by vkotronis** [comment]
@slowr the codefactor complexity warning is not due to the newly added code. Old (complex) methods will be refactored in separate issue ( #377 ). So you can review based on the current state (the PR is stable on my side, and tested).

**Pull request merged**